src/test/scala/pl/project13/kaze/KazeClassSpec.scala

```diff
@@ -43,13 +62,66 @@ class KazeClassSpec extends WordSpec with Matchers {
            |      timeout = timeout)
            |
            |  override def toString =
-           |    s```Person($name,$age,$item,$items,$opt,$timeout)```
+           |    s```Person(name=$name,age=$age,item=$item,items=$items,opt=$opt,timeout=$timeout)```
+           |
+           |  override def equals(other: Any): Boolean = other match {
+           |    case that: Person =>
+           |      java.util.Objects.equals(this.name, that.name) &&
+           |      java.util.Objects.equals(this.age, that.age) &&
+           |      java.util.Objects.equals(this.item, that.item) &&
+           |      java.util.Objects.equals(this.items, that.items) &&
+           |      java.util.Objects.equals(this.opt, that.opt) &&
+           |      java.util.Objects.equals(this.timeout, that.timeout)
+           |        case _ => false
+           |  }
+           |
+           |  override def hashCode(): Int =
+           |    java.util.Objects.hash(
+           |      name,
+           |      Int.box(age),
+           |      item,
+           |      items,
+           |      opt,
+           |      timeout)
            |}
            |object Person {
            |  /** Scala API */
-           |  def apply() = new Person()
+           |  def apply(): Person = new Person()
            |  /** Java API */
-           |  def getInstance() = apply()
+           |  def getInstance(): Person = apply()
+           |  /** Scala API */
+           |  def apply(
+           |    name: String,
+           |    age: Int,
+           |    item: Item,
+           |    items: List[Item],
+           |    opt: Option[String],
+           |    timeout: scala.concurrent.duration.FiniteDuration
+           |  ): Person = new Person(
+           |    name,
+           |    age,
+           |    item,
+           |    items,
+           |    opt,
+           |    timeout
+           |  )
+           |
+           |  /** Java API */
+           |  def create(
```

Comment:
since this is create the other one should be to i suppose

Reply:
This one makes things more duplicated and harder to evolve when you add another field; can we remove this? the intended  style is to use the with... overloads